Documentation/building/windows/dependencies.md

```diff
@@ -0,0 +1,44 @@
+# Build Dependencies for Windows
+
+Building Xamarin.Android requires:
+
+  * An existing installation of the Xamarin.Android SDK and the Android SDK
+  * The .NET Framework 3.5 &ndash; 4.7 development tools
```

Comment:
Wouldn't it be '.NET Framework v4.7.2'?  I'm not sure about the appropriate terminology.

Reply:
It is the ".NET Framework 3.5" part which confuses me; .NET 3.5 is *ancient*.

Again, I don't know offhand what the appropriate names/terminology should be -- though *presumably* we'd want/need the [.NET Framework 4.7 Developer Pack](https://www.microsoft.com/en-us/download/details.aspx?id=55168)? -- but the "3.5" part threw me for a loop, at least.

We should ensure that we're using the correct names, regardless, and ideally in a manner which doesn't start with "Install Visual Studio...".

Reply:
### About .NET Framework 3.5

Here are a few of the places .NET Framework 3.5 is currently still used:

* *xamarin-android/external/Java.Interop/external/cecil/Mono.Cecil.props*
* *xamarin-android/external/opentk/Source/Bind/Generator.Bind.csproj*
* *xamarin-android/external/opentk/Source/Converter/Generator.Convert.csproj*
* *xamarin-android/external/mono/external/cecil/Mono.Cecil.props*
* *xamarin-android/external/mono/external/linker/cecil/Mono.Cecil.props*

If I recall correctly, I think it was *xamarin-android/external/Java.Interop/external/cecil/Mono.Cecil.props* or *xamarin-android/external/mono/external/cecil/Mono.Cecil.props* that caused the first build error when the .NET Framework 3.5 development tools weren't installed, so I'm not sure how many of the other occurrences of the target framework would eventually be used by the build process.

### About command line installation steps to work without the Visual Studio Installer

If we want a set of instructions that will work outside of the Visual Studio Installer, I can prepare a set like that too. Then we'd have a set of instructions for folks who are using Visual Studio, and a set for folks who'd rather just install the command line tools. I think it's still useful to include the instructions for the Visual Studio Installer in that case.

Reply:
To jump in on the the .Net Framework 3.5 discussion, I run into an error when building `r8.csproj` on Windows if this is not installed.
```
"C:\Users\peter\source\pj\xamarin-android\src\r8\r8.csproj" (default target) (28:2) ->
(GetReferenceAssemblyPaths target) ->
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3645: .NET Framework v3.5 Service Pack 1 was not found. In order to targe
t ".NETFramework,Version=v2.0", .NET Framework v3.5 Service Pack 1 or later must be installed. [C:\Users\peter\source\pj\xamarin-android\src\r8\r8.csproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3644: The reference assemblies for framework ".NETFramework,Version=v2.0"
 were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Not
e that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [C:\Us
ers\peter\source\pj\xamarin-android\src\r8\r8.csproj]
```

I also see a related prompt when loading Xamarin.Android.sln into VS2017:
![image](https://user-images.githubusercontent.com/2000163/52436407-36b02600-2ae2-11e9-9d48-2541f7c3d9da.png)

These particular issues are likely addressed by https://github.com/xamarin/xamarin-android/pull/2666 however.